File: app/src/main/java/com/example/myapplication/ViewModel/AccessoryViewModel.java
package com.example.myapplication.ViewModel;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.myapplication.NetworkService;

import java.util.ArrayList;

import Models.linked.Product;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
public class AccessoryViewModel extends ViewModel {
    MutableLiveData<ArrayList<Product>> AccessoryLiveData;
    private String TYPE = "Accessory";
    public AccessoryViewModel()
    {
        AccessoryLiveData= new MutableLiveData<>();
        init();
    }
    public MutableLiveData<ArrayList<Product>> getAccessoryLiveData()
    {
        return AccessoryLiveData;
    }
    public void init()
    {
        NetworkService.getInstance().getApi().GetCatalog(TYPE).enqueue(new Callback<ArrayList<Product>>() {
            @Override
            public void onResponse(Call<ArrayList<Product>> call, Response<ArrayList<Product>> response) {
                AccessoryLiveData.postValue(response.body());
            }
            @Override
            public void onFailure(Call<ArrayList<Product>> call, Throwable t) {

            }
        });
    }
}

File: app/src/main/java/com/example/myapplication/ViewModel/AccessoryViewModel.kt
package com.example.myapplication.ViewModel

import com.example.myapplication.NetworkService.Companion.instance
import androidx.lifecycle.ViewModel
import androidx.lifecycle.MutableLiveData
import Models.linked.Product
import com.example.myapplication.domain.repositories.AccessoriesRepository
import com.example.myapplication.domain.usecases.GetAccessoryUseCase
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.util.ArrayList

class AccessoryViewModel(accessoryUseCase: GetAccessoryUseCase) : ViewModel() {
    var AccessoryUseCase : GetAccessoryUseCase
    init{
        AccessoryUseCase=accessoryUseCase
    }
    var accessoryLiveData: MutableLiveData<ArrayList<Product>>
    fun getAccessories(skip : Int, count : Int)
    {
        CoroutineScope(Dispatchers.IO).launch {
            accessoryLiveData.postValue(AccessoryUseCase.Execute(skip, count))
        }
    }
    init {
        accessoryLiveData = MutableLiveData()
    }
}